File: Python/f/f3-Calculadora.py
'''
4. Implementa una calculadora básica
'''
def suma(a, b):
    return int(a) + int(b)

def resta(a, b):
    return int(a) - int(b)

def multiplicacion(a, b):
    return int(a) * int(b)

def division(a, b):
    if int(b) == 0:
        return "No se puede dividir por 0"
    return int(a) / int(b)

op = int( input("Ingrese el primer número: ") )

match op:
    case 1:
        print("Suma\n-------------------------------")
        s1 = input("Introduce el primer número: ")
        s2 = input("Introduce el segundo número: ")
        print(suma(s1, s2))
    case 2:
        print("Resta\n-------------------------------")
        r1 = input("Introduce el primer número: ")
        r2 = input("Introduce el segundo número: ")
        print(resta(r1, r2))
    case 3:
        print("Multiplicacion\n-------------------------------")
        m1 = input("Introduce el primer número: ")
        m2 = input("Introduce el segundo número: ")
        print(multiplicacion(m1, m2))
    case 4:
        print("Division\n-------------------------------")
        d1 = input("Introduce el primer número: ")
        d2 = input("Introduce el segundo número: ")
        print(division(d1, d2))

File: Rust/f/f3-Calculadora.rs
// 3. Programa una calculadora basica.
use std::io; // Necesario para la entrada/salida estándar

fn input() -> i32 {
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Por favor introduce un número");
    input.trim().parse().expect("Por favor introduce un número")
}
fn main(){
    //  Menu
    println!("1.Suma");
    println!("2.Resta");
    println!("3.Multiplicacion");
    println!("4.Division\nIntroduce el número de la operación: ");

    let mut n = String::new();
    let _ = io::stdin().read_line(&mut n);
    let op = n.trim().parse().expect("Por favor introduce un número");
    match op {
        1 => { println!("Suma\n-------------------------------");
        println!("Introduce el primer número: ");
        let s1:i32 = input();
        println!("Introduce el segundo número: ");
        let s2:i32 = input();
        println!("{} + {} es {}", s1, s2, s1+s2);
        },
        2 => { println!("Resta\n-------------------------------");
        println!("Introduce el primer número: ");
        let s1:i32 = input();
        println!("Introduce el segundo número: ");
        let s2:i32 = input();
        println!("{} - {} es {}", s1, s2, s1-s2);
        },
        3 => { println!("Multiplicacion\n-------------------------------");
        println!("Introduce el primer número: ");
        let m1:i32 = input();
        println!("Introduce el segundo número: ");
        let m2:i32 = input();
        println!("{} * {} es {}", m1, m2, m1*m2);
        },
        4 => { println!("Division\n-------------------------------");
        println!("Introduce el primer número: ");
        let d1:i32 = input();
        println!("Introduce el segundo número: ");
        let d2:i32 = input();
        println!("{} / {} es {}", d1, d2, d1/d2);
        },
        _ => println!("Operación no reconocida"),
    }



}
